### ISSUE TITLE
Port JSCS rules for: requireSpacesInFunction, etc.

### ISSUE THREAD

**Issue by meister** [eslint/eslint#6099]
**When does this rule warn? Please describe and show example code:**
The rule allows enforcing or preventing spaces after function parens and before curly bracers. Currently it’s possible to control the space between keyword and parenthesis (`function ()`), but not after keyword+parenthesis. Examples:
```javascript
function doSomething(){
//                    ^--- Missing/extra space not checked
    return;
}

for (var i = 0; i < items.length; i++){
//                                    ^--- Missing/extra space not checked
}

try {
    doSomething();
} catch (err){
//           ^--- Missing/extra space not checked
    throw new Error('message');
}

switch (local){
//            ^--- Missing/extra space
  case 1:
    doSomething();
}
```

**Is this rule preventing an error or is it stylistic?**
Stylistic

**Why is this rule a candidate for inclusion instead of creating a custom rule?**
Compatibility with JSCS:
```
{
	"requireSpacesInAnonymousFunctionExpression": {
		"beforeOpeningCurlyBrace": true
	},
	"requireSpacesInFunctionDeclaration": {
		"beforeOpeningCurlyBrace": true
	},
	"requireSpacesInFunction": {
		"beforeOpeningCurlyBrace": true
	}
}
```

**Are you willing to create the rule yourself?**

**Comment by mysticatea** [comment]
Thank you for this issue.

Is [space-before-blocks](http://eslint.org/docs/rules/space-before-blocks) rule not enough?

**Comment by meister** [comment]
Wow.. I thought I had checked all the rules. I guess I hadn’t – I can indeed use that rule for this specific use case. Closing my issue.

**Comment by mysticatea** [comment]
FYI: There is a table to identify gap between JSCS and ESLint: https://github.com/eslint/eslint/issues/5856#issuecomment-216484984

**Comment by meister** [comment]
Thanks so much for feedback and help. Would you also have any idea about the progress of https://github.com/eslint/eslint/issues/4680?

It’s the last rule for our company’s requirements. There doesn’t seem to be a good way to port the rules for code like this:

```
// Required one-var for consecutive declarations
var a = 1,
    b = 2,
    items = [];

// JSCS throws a warning for rule 'requireMultipleVarDecl:true'
// ESLint throws a warning for rule 'one-var'
var c = a;

doSomething(a, b);

// JSCS does not throw a warning, since a statement was in between var declarations
// ESLint throws a warning
var d = c + b;

// Another block with var
// JSCS does not throw a warning for i
// ESLint throws a warning for i
for (var i = 0; i < items.length; i++) {
    // Another block of consecutive vars
    // JSCS does not throw a warning
    // ESLint throws a warning
    var temp1 = a + c,
        temp2 = b + d;

	doSomething(temp1, temp2);
}
```

**Comment by mysticatea** [comment]
I'm not sure.
I feel that the responsibility of  `one-var` and `vars-on-top` is overlapping.

**Comment by meister** [comment]
Yes, that was exactly my concern. If `one-var` was more relaxed (like JSCS), but `vars-on-top` was enforced, the outcome would be almost exactly how it is now (all vars on top, with 1 var declaration). Otherwise, separated blocks of vars (if there are statements in the middle) would be allowed.
